lib_lua/src/luaconf.h

```diff
@@ -515,7 +515,7 @@
 */
 #define LUAI_UACNUMBER	double
 
-#define LUA_NUMBER_PRECISION 3
+#define LUA_NUMBER_PRECISION 6
```

Comment:
In reality, the vast majority of the time we're printing out variables other than Lat/Log (CAN bus values, debug values, analog channel values, IMU values, etc. This will force all output to 6 decimals of precision, causing a usability issue with scripts.

I think we can have the 6 precision if we modify the ftoa conversion to strip trailing zeros (and the decimal point if needed). That way, we can have the best of both worlds - full precision when we need it, and concise output when possible.

Bonus: have lua use our common ftoa source